File: MATLAB/P04.m
flag = 1;

while flag == 1
    
    fprintf('\n');
	fprintf('####################################################################\n');
	fprintf('# 1. MOSTRAR LA SUMATORIA DESDE 0 HASTA UN NÚMERO n ---------- (c) #\n');
	fprintf('# 2. SALIR --------------------------------------------------- (s) #\n');
	fprintf('####################################################################\n');
	fprintf('\n');

	opt = input('INGRESE LA OPCIÓN QUE DESEA EJECUTAR: ','s');
	fprintf('\n');	                
    
    switch opt
        
        case 'c'
            
            num = input('INGRESE UN NÚMERO n: ');
            suma = 0;
            for i = 0:num
                
                suma = suma + i;
            
            end
        
        fprintf('LA SUMATORIA DEL NÚMERO %g ES: %g',num,suma);    
            
        case 's'
            
            flag = 0;
        
        otherwise
            
            fprintf('######################################\n');
            fprintf('# USTED NO INGRESO UNA OPCIÓN VALIDA #\n');
            fprintf('######################################\n');
    end
    
    fprintf('\n');
    
end

fprintf('####################################\n');
fprintf('# GRACIAS POR UTILIZAR EL PROGRAMA #\n');
fprintf('####################################\n');

File: PYTHON/P04.py
flag = True

while flag == True:

	try: 

		archivo = open("Salida_P04.txt","a")
		print("")
		print("##############################################################")
		print("# 1. MOSTRAR LA SUMATORIA DESDE 0 HASTA UN NÚMERO n ---- (n) #")
		print("# 2. SALIR --------------------------------------------- (s) #")
		print("##############################################################")
		print("")

		opt = input("INGRESE LA OPCIÓN QUE DESEA EJECUTAR: ")
		print("")

		if opt == "n":
			
			suma = int(0)
			n = int(input("INGRESE UN NÚMERO n: "))

			for i in range(0,n+1):

				suma += i

			print("LA SUMATORIA DEL NÚMERO ",n," ES: ",suma)
			archivo.write('LA SUMATORIA DEL NÚMERO %s'%(n)+' ES: %s'%suma+'\n')
			archivo.close()

		elif opt == "s":

			flag = False

		else: 

			print("######################################")
			print("# USTED NO INGRESO UNA OPCIÓN VALIDA #")
			print("######################################")

	except:

		print("USTED NO INGRESÓ UN NÚMERO ENTERO POSITIVO.")

print("")
print("####################################")
print("# GRACIAS POR UTILIZAR EL PROGRAMA #")
print("####################################")